src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisher.java

```diff
@@ -124,12 +127,27 @@ GHCheckRunBuilder getUpdater(final GitHub github, final GitHubChecksDetails deta
     @VisibleForTesting
     GHCheckRunBuilder getCreator(final GitHub gitHub, final GitHubChecksDetails details) throws IOException {
         GHCheckRunBuilder builder = gitHub.getRepository(context.getRepository())
-                .createCheckRun(details.getName(), context.getHeadSha())
-                .withStartedAt(details.getStartedAt().orElse(Date.from(Instant.now())));
+            .createCheckRun(details.getName(), context.getHeadSha())
+            .withStartedAt(details.getStartedAt().orElse(Date.from(Instant.now())));
 
         return applyDetails(builder, details);
     }
 
+    @VisibleForTesting
+    StandardUsernameCredentials getCredentials() {
+        if (credentials == null) {
+            credentials = context.getCredentials();
+            if (credentials instanceof GitHubAppCredentials) {
+                final var gitHubAppCredentials = (GitHubAppCredentials) credentials;
+                if (context instanceof GitHubSCMSourceChecksContext) {
+                    final var gitHubSCMSourceChecksContext = (GitHubSCMSourceChecksContext) context;
+                    credentials = gitHubAppCredentials.withOwner(gitHubSCMSourceChecksContext.getOwner());
```

Comment:
FWIW, I think it is undesirable to have the `withOwner` method be considered a public API of the plugin. Is there a reason we cannot use `Connector.lookupScanCredentials` from `github-branch-source` instead to handle contextualization automatically? It looks like it would require some API changes to `SCMFacade` methods to have them accept an `owner` parameter, but would that be ok, or is there some fundamental problem with it?

Reply:
Sounds fine if someone wants to work on it.

Reply:
I'll create a follow-up PR ASAP.